devsnest/app/controllers/api/v1/admin/projects_controller.rb

```
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    module Admin
+      # Devsnest Projects
+      class ProjectsController < ApplicationController
+        include JSONAPI::ActsAsResourceController
+        before_action :admin_auth
+        before_action :check_challenge_type, only: %i[create]
+        before_action :check_if_project_exists, only: %i[create]
```

we can merge these both functions into one function, and these checks should be in the model and we should return the errors if there are any from the controller